app/api/events.py

```diff
@@ -858,7 +859,7 @@ def query(self, view_kwargs):
                         Event.event_type_id != None,
                         Event.event_topic_id != None,
                         Event.event_sub_topic_id != None,
-                        Event.tickets.any(and_(Ticket.deleted_at == None, Ticket.is_hidden == False, Ticket.sales_ends_at > current_time, Ticket.is_available == True)),
+                        Event.tickets.any(and_(Ticket.deleted_at == None, Ticket.is_hidden == False, Ticket.sales_ends_at > current_time, Ticket.quantity > get_sold_and_reserved_tickets_count(Ticket.id))),
```

Comment:
Won't work. This is not a python loop, only allowed operations here are sqlalchemy scalar and join type operations

Reply:
Please verify by running before submitting

Reply:
it's working, I have checked it

Reply:
I have an event, it has two tickets and 1 ticket of each type is sold.then after applying above I edited event and changes ticket number to 1 of both events then the event is not on the front page anymore. Then i changed ticket numbers for one ticket to one and for another ticket to 100. Then event is showing on front page.

Reply:
Hmm, yes, it'll work indeed. But I'll have to check if the query is efficient. Sorry for not noticing `Ticket.id` instead of `ticket.id`

Reply:
Doesn't work, it statically returns all reserved and sold tickets and compares ticket quantity with it

Reply:
`get_sold_and_reserved_tickets_count(Ticket.id)` what is meaning of passing `ticked.id` here?

Reply:
`all reserved and sold tickets ` for a ticket ?

Reply:
It won't work because it expects a ticket ID. Passing `Ticket.id` just returns all tickets sold count